### DISCUSSION TITLE
using keka in crontab or shell script (automator)

### DISCUSSION THREAD

**Discussion by zero7404** [aonez/Keka#1038] [Q&A]
i have Keka and would like to use it in one of these 2 ways to automate my archives.  i currently have automator workflows for my archive jobs but that uses the default apple archive utility, and i would like to end up with uncompressed tar archives.

for example, i want to archive my ios device backup from terminal:
`% Keka --ignore-file-access -cli tar -cf /Users/zero7404/Documents/iOS\ Archives/zero7404\ iphone\ archive.tar /Users/zero7404/Library/Application\ Support/MobileSync/Backup/00008110-000174E83682801E`

but terminal says:
`zsh: command not found: Keka`

i was hoping i could get this command into crontab, and i've tried the traditional method of creating a shell script (that works when run in terminal), for some reason when i add it in crontab, it will not run.  i use full path names:

`#!/bin/sh`
`export PATH=$PATH:/usr/bin`
`rm -rv /Users/zero7404/Documents/iOS\ Archives//zero7404\ iphone\ archive.tar`
`sleep 1s`
`tar -cvf /Users/zero7404/Documents/iOS\ Archives/zero7404\ iphone\ archive.tar /Users/zero7404/Library/Application\ Support/MobileSync/Backup/00008110-000174E83682801E/*`

whether i can do something like this in automator or directly in crontab, i would be happy with either.  if i create an app using automator, i insert it into crontab with the 'open -a' command and set my other parameters.

for example, my current crontab looks like this and these apps (created using automator workflows) execute fine at the designated times:

`10 11 * * 1,3,5 open -a zero7404\ iphone\ archive.app`
`20 11 * * 1,3,5 open -a apple\ mail\ archive.app`
`30 11 * * 1,3,5 open -a apple\ emlx\ archive.app`

if you would be able to assist me in any way I would greatly appreciate that.

**Comment by aonez** [comment]
Note that unless you export the path of the Keka binary, you should call it like this:

```sh
/Applications/Keka.app/Contents/MacOS/Keka ...
```

Your line:

```sh
% /Applications/Keka.app/Contents/MacOS/Keka --ignore-file-access -cli tar -cf /Users/zero7404/Documents/iOS\ Archives/zero7404\ iphone\ archive.tar /Users/zero7404/Library/Application\ Support/MobileSync/Backup/00008110-000174E83682801E
```

Obviously this is in case you have Keka in your Applications folder. Replace with your used folder if needed.

**Comment by zero7404** [comment]
thank you for the advice aonez ... really appreciate it

can i insert a line like this in crontab to have it create my tar every day at 11 am ?

`0 11 * * * /Applications/Keka.app/Contents/MacOS/Keka --ignore-file-access -cli tar -cf /Users/zero7404/Documents/iOS\ Archives/zero7404\ iphone\ archive.tar /Users/zero7404/Library/Application\ Support/MobileSync/Backup/00008110-000174E83682801E`

Keka is already granted Full Disk Access, but it is not listed in Accessibility (control your computer).

concern here is that when it's launched from crontab, it will run with the correct user privileges.

**Comment by aonez** [comment]
I'm not using `crontab` for ages (used it on Linux) so I can't assure you, but you can simply try it and let me know :)

**Reply by zero7404** [reply]
just looking for confirmation that the command for Keka executable starts with:

/Applications/Keka.app/Contents/MacOS/Keka ...

i don't think i need to insert an 'open' command to the left of it as you would do when trying to run a .app file from within crontab.

**Comment by zero7404** [comment]
here is what i inserted in crontab:

`0 10 * * * /Applications/Keka.app/Contents/MacOS/Keka --ignore-file-access -cli tar -cf /Users/zero7404/Documents/iOS\ Archives/zero7404\ iphone\ archive.tar /Users/zero7404/Library/Application\ Support/MobileSync/Backup/00008110-000174E83682801E/*`

it runs Keka, but this comes up, it doesn't appear to follow the tar command that's stated in the crontab line ...

[https://drive.google.com/file/d/1zzMDHHAq3KMHYPMYbxr7NIugu9dxIDPs/view?usp=sharing](url)

here is the dialog of the error that the keka progress bar shows:

> OS: Version 11.6.3 (Build 20G415) (x86_64)
> Keka: v1.2.52-r4882 (WEB) (Sandboxed) (en)
> Binary used: keka7zz
> Arguments: (
>     a,
>     "-t7z",
>     "-snh",
>     "-snl",
>     "-mx0",
>     "-ms=on",
>     "-xr!.DS_Store",
>     "-xr!.localized",
>     "-xr!._*",
>     "-xr!.FBC*",
>     "-xr!.Spotlight-V100",
>     "-xr!.Trash",
>     "-xr!.Trashes",
>     "-xr!.background",
>     "-xr!.TemporaryItems",
>     "-xr!.fseventsd",
>     "-xr!.com.apple.timemachine.*",
>     "-xr!.VolumeIcon.icns",
>     "-bsp1",
>     "/Users/zero7404/Documents/iOS Archives/Archive.7z",
>     "./ar",
>     "./Users/zero7404/Library/Application Support/MobileSync/Backup/00008110-000174E83682801E/*"
> )
> 
> 7-Zip (z) 21.07 (x64) : Copyright (c) 1999-2021 Igor Pavlov : 2021-12-26 : Modified by aone for Keka
>  64-bit locale=en_US.UTF-8 Threads:12
> 
> Scanning the drive:
>   0M Scan 
> 
> WARNING: 
>   0M Scan  ./Users/zero7404/Library/Appli . ckup/00008110-000174E83682801E/
> errno=2 : No such file or directory
> ./ar
> 
> 
> 0 files, 0 bytes
> 
> 
> 
> WARNING: errno=1 : Operation not permitted
> ./Users/zero7404/Library/Application Support/MobileSync/Backup/00008110-000174E83682801E/
> 
> 
> Creating archive: /Users/zero7404/Documents/iOS Archives/Archive.7z
> 
> Add new data to archive: 0 files, 0 bytes
> 
>   0%
> 
> 
> Files read from disk: 0
> Archive size: 32 bytes (1 KiB)
> 
> Scan WARNINGS for files and folders:
> 
> ./ar : errno=2 : No such file or directory
> ./Users/zero7404/Library/Application Support/MobileSync/Backup/00008110-000174E83682801E/ : errno=1 : Operation not permitted
> ----------------
> Scan WARNINGS: 2
> 
> Error code 1

**Comment by aonez** [comment]
Sorry @zero7404 I did not noticed you're using `-cli`, it is `--cli`.

**Reply by zero7404** [reply]
thanks for spotting that.

added cron to full disk access in sysprefs. after that, the regular tar command ran successfully at the designated time.

if some added benefit to using the keka option over standard tar in this manner, willing to try it.
